pyperformance/_pip.py

```diff
@@ -150,7 +150,7 @@ def install_requirements(reqs, *extra,
     if upgrade:
         args.append('-U')  # --upgrade
     for reqs in [reqs, *extra]:
-        if os.path.exists(reqs):
+        if os.path.isfile(reqs):
             args.append('-r')  # --requirement
         args.append(reqs)
```

Comment:
As suggested above, the fact that you can't pass the full setup.py filename to pip should probably be handled here.  So something like:

```python
    for req in [reqs, *extra]:
        if os.path.isfile(req):
            name = os.path.basename(req)
            if re.match(r'^setup.*\.py$', name):
                if name != setup.py:
                    raise ValueError(f'pip doesn't support installing from {req!r}')
                req = os.path.dirname(req)
            else:
                args.append('-r')  # --requirement
        args.append(req)
```

Again, I'm not going to block merging on this.

Reply:
Yeah -- I agree it makes sense to move the handling of this to here.  I'm not sure I follow the regex suggestion, though.  Are there cases where the filename wouldn't literally be "setup.py"?

Reply:
In any case, I moved this logic here, as you suggested, but just kept it to a direct string comparison to `setup.py`.  Let me know if the regex makes sense, otherwise, I think this is good to go.

Reply:
Yeah, the regex is overkill for a less-than-likely hypothetical.  If we need to support more than setup.py, we can do that later.

So:

```python
    for req in [reqs, *extra]:
        if os.path.isfile(req):
            if os.path.basename(req) == 'setup.py':
                req = os.path.dirname(req)
            else:
                args.append('-r')  # --requirement
        args.append(req)
```